tbpore/tbpore.py

```diff
@@ -111,35 +195,113 @@ def main(
         name = inputs[0].name.split(".")[0]
         if not name:
             name = "input"
-
         logger.debug(f"No sample name found; using '{name}'")
 
     infile = tmp / f"{name}.fq.gz"
-    if len(inputs) == 1 and inputs[0].is_file():
-        if not inputs[0].suffix == ".gz":
-            with open(inputs[0], "rb") as f_in:
-                with gzip.open(infile, "wb") as f_out:
-                    shutil.copyfileobj(f_in, f_out)
-        else:
-            shutil.copy2(inputs[0], infile)
+    concatenate_inputs_into_infile(inputs, infile, recursive, ctx)
 
-    else:
-        logger.info("Searching for fastq files...")
-        fq_files = []
-        for p in inputs:
-            if p.is_file():
-                fq_files.append(p)
-            else:
-                fq_files.extend(find_fastq_files(p, recursive))
-        logger.info(f"Found {len(fq_files)} fastq files. Joining them...")
-        concatenate_fastqs(fq_files, infile)
+    report_all_mykrobe_calls_param = "-A" if report_all_mykrobe_calls else ""
+    mykrobe_output = f"{outdir}/{name}.mykrobe.json"
+    mykrobe = ExternalTool(
+        tool="mykrobe",
+        input=f"-i {infile}",
+        output=f"-o {mykrobe_output}",
+        params=f"predict {report_all_mykrobe_calls_param} --sample {name} -t {threads} --tmp {tmp} "
+        f"--skeleton_dir {cache_dir} {config['mykrobe']['predict']['params']}",
+        logdir=logdir,
+    )
+
+    subsampled_reads = f"{tmp}/{name}.subsampled.fastq.gz"
+    rasusa = ExternalTool(
+        tool="rasusa",
+        input=f"-i {infile}",
+        output=f"-o {subsampled_reads}",
+        params=config["rasusa"]["params"],
+        logdir=logdir,
+    )
+
+    sam_file = f"{tmp}/{name}.subsampled.sam"
+    minimap = ExternalTool(
+        tool="minimap2",
+        input=f"{H37RV_genome} {subsampled_reads}",
+        output=f"-o {sam_file}",
+        params=f"-t {threads} {config['minimap2']['params']}",
+        logdir=logdir,
+    )
+
+    sorted_sam_file = f"{tmp}/{name}.subsampled.sorted.sam"
+    samtools_sort = ExternalTool(
+        tool="samtools",
+        input=sam_file,
+        output=f"-o {sorted_sam_file}",
+        params=f"sort -@ {threads} {config['samtools']['sort']['params']}",
+        logdir=logdir,
+    )
+
+    pileup_file = f"{tmp}/{name}.subsampled.pileup.vcf"
+    bcftools_mpileup = ExternalTool(
+        tool="bcftools",
+        input=sorted_sam_file,
+        output=f"-o {pileup_file}",
+        params=f"mpileup -f {H37RV_genome} --threads {threads} {config['bcftools']['mpileup']['params']}",
+        logdir=logdir,
+    )
+
+    snps_file = f"{tmp}/{name}.subsampled.snps.vcf"
+    bcftools_call = ExternalTool(
+        tool="bcftools",
+        input=pileup_file,
+        output=f"-o {snps_file}",
+        params=f"call --threads {threads} {config['bcftools']['call']['params']}",
+        logdir=logdir,
+    )
+
+    filtered_snps_file = f"{outdir}/{name}.subsampled.snps.filtered.vcf"
```

Comment:
```suggestion
    filtered_snps_file = f"{outdir}/{name}.subsampled.snps.filtered.bcf"
```

Probably better to make this a BCF file to save disk space